dapp/src/components/AccountListener.js

```diff
@@ -14,7 +14,10 @@ const AccountListener = (props) => {
   const web3react = useWeb3React()
   const { account, chainId, library } = web3react
   const prevAccount = usePrevious(account)
+  const [contracts, setContracts] = useState(null)
+  const [balancesInterval, setBalancesInterval] = useState(null)
```

Comment:
Any reason to have it as a state and not a local variable inside `useEffect`?

Reply:
Yup your right cleaning that up